crates/transaction-pool/src/traits.rs

```diff
@@ -207,6 +207,13 @@ pub trait TransactionPool: Send + Sync + Clone {
     ) -> Vec<Arc<ValidPoolTransaction<Self::Transaction>>>;
 }
 
+/// Extension for [TransactionPool] trait that allows to set the current block info.
+#[auto_impl::auto_impl(Arc)]
+pub trait TransactionPoolExt: TransactionPool {
```

Comment:
I'll think about naming a bit, not perfectly happy with `Ext`, but is sufficient.

Reply:
@mattsse i didn't want to give it any other name until we define what these supplementary methods are. but, perhaps, `TransactionPoolMut` would be better since we are somehow modifying the pool inner state